pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -438,7 +440,12 @@ stdenv.mkDerivation {
     "-Dsmack=true"
     "-Db_pie=true"
     "-Dinstall-sysconfdir=false"
-    "-Defi-ld=${stdenv.cc.bintools.targetPrefix}ld"
+    "-Defi-ld=gold"
```

Comment:
Note: systemd now only supports either `gold` or `bfd` as value here. We should test  cross compiling.

Reply:
This definitely breaks cross-compilation without any target prefix. We might need a wrapper when systemd does support a full name here.

Reply:
Since those values are directly passed to binutils, they should not break when cross compiling: https://github.com/systemd/systemd/pull/21264/files#diff-ba74c03f6f96be2712bae81ef72761cea2ec521483d82e64e1eb174809eb84a2R259